app/device/gps.py

```diff
@@ -1,34 +1,37 @@
-import Adafruit_BBIO.UART as UART
 from time import sleep
 import serial
+import logging
 
 class GPS(object):
     def __init__(self):
-        UART.setup("UART1")
-        self.ser = serial.Serial(port = '/dev/ttyO1', baudrate=9600)
-        self.set_baud_rate(57600)
+        self.ser = serial.Serial(port = '/dev/ttyAMA0', baudrate=9600)
         self.set_data_rate(0.2)
         self.set_datatype("GPRMC_GPGGA")
 
         sleep(1)
         self.ser.flushInput()
-        print "GPS Initialized"
+        logging.info('GPS Initialized')
+
 
     def read(self):
-        print "Starting read"
         self.ser.flushInput()
 
         data1 = self.ser.readline().split(',')
         data2 = self.ser.readline().split(',')
+        raw_data = {}
 
         if data1[0] == "$GPRMC":
             raw_data["GPRMC"] = data1[1:]
             raw_data["GPGGA"] = data2[1:]
         else:
             raw_data["GPRMC"] = data2[1:]
             raw_data["GPGGA"] = data1[1:]
+        try:
+            return GPS.parse_raw_data(raw_data)
+        except Exception as e:
+            logging.error('error: {}, raw_data: {}'.format(e, raw_data))
```

Comment:
if there are cases where we can continue in the event of an exception...we should always try to log it and return a reasonable value